minecraft/1.7/src/main/java/nova/core/wrapper/mc/forge/v17/asm/lib/ComponentInjector.java

```diff
@@ -60,7 +60,8 @@ public ComponentInjector(Class<T> baseClazz) {
 		this.baseClazz = baseClazz;
 	}
 
-	public synchronized T inject(ComponentProvider provider, Class<?>[] typeArgs, Object[] args) {
+	@SuppressWarnings("rawtypes")
+	public synchronized T inject(ComponentProvider<? extends ComponentProvider<? extends ComponentProvider.ComponentMap>.ComponentMap> provider, Class<?>[] typeArgs, Object[] args) {
```

Comment:
This is what happens when you have a non-static class that you pass as a type to the class that contains said non-static class